gammapy/utils/fitting/tests/test_iminuit.py

```diff
@@ -7,17 +7,19 @@
 pytest.importorskip("iminuit")
 
 
-def fcn(parameters):
-    x = parameters["x"].value
-    y = parameters["y"].value
-    z = parameters["z"].value
-    x_opt, y_opt, z_opt = 2, 3e5, 4e-5
-    x_err, y_err, z_err = 0.2, 3e4, 4e-6
-    return (
-        ((x - x_opt) / x_err) ** 2
-        + ((y - y_opt) / y_err) ** 2
-        + ((z - z_opt) / z_err) ** 2
-    )
+class MyDataset:
```

Comment:
Derive from the `Dataset` base class?
(Is this a duplicated test case? Can one variant be removed?)

Reply:
Yes, this is partly a duplicated test case. The fitting backends are tested individually and via the `Fit` API again in `utils/fitting/test_fit.py`. I agree this should be refactored, I'll make a reminder issue for v0.13.